src/Estimators/src/SubModelKinDynWrapper.cpp

```diff
@@ -0,0 +1,327 @@
+            return false;
+        }
+
+        // Update dJnu
+        m_dJnuList[m_subModel.getAccelerometerList()[idx].name]
+            = iDynTree::toEigen(m_kinDyn.getFrameBiasAcc(m_subModel.getAccelerometerList()[idx].frame));
+
+        // Update rotMatrix
+        iDynTree::Transform H
+            = m_kinDyn.getWorldTransform(m_subModel.getAccelerometerList()[idx].frame);
+        m_accRworldList[m_subModel.getAccelerometerList()[idx].name] = iDynTree::toEigen(H.getRotation().inverse());
+    }
+
+    // Update gyroscope jacobians
+    for (int idx = 0; idx < m_subModel.getGyroscopeList().size(); idx++)
+    {
+        if (!m_kinDyn.getFrameFreeFloatingJacobian(m_subModel.getGyroscopeList()[idx].frame,
+                                                 m_jacGyroList[m_subModel.getGyroscopeList()[idx].name]))
+        {
+            blf::log()->error("{} Unable to get the compute the free floating jacobian of the "
+                              "frame {}.",
+                              logPrefix,
+                              m_subModel.getGyroscopeList()[idx].frame);
+            return false;
+        }
+    }
+
+    // Update external contact jacobians
+    for (int idx = 0; idx < m_subModel.getExternalContactList().size(); idx++)
+    {
+        if (!m_kinDyn.getFrameFreeFloatingJacobian(m_subModel.getExternalContactList()[idx],
+                                                 m_jacContactList[m_subModel
+                                                                    .getExternalContactList()[idx]]))
+        {
+            blf::log()->error("{} Unable to get the compute the free floating jacobian of the "
+                              "frame {}.",
+                              logPrefix,
+                              m_subModel.getExternalContactList()[idx]);
+            return false;
+        }
+    }
+
+    return true;
+}
+
+bool RDE::SubModelKinDynWrapper::inverseDynamics(Eigen::VectorXd& motorTorqueAfterGearbox,
+                                                 Eigen::VectorXd& frictionTorques,
+                                                 Eigen::VectorXd& tauExt,
+                                                 Eigen::VectorXd& baseAcceleration,
+                                                 Eigen::VectorXd& jointAcceleration)
+{
+    //    auto massMatrixEigen(iDynTree::toEigen(massMatrix));
+    long unsigned int nrOfJoints = m_subModel.getModel().getNrOfDOFs();
+    Eigen::MatrixXd FTranspose(m_massMatrix.block(6, 0, nrOfJoints, 6));
+    Eigen::MatrixXd H(m_massMatrix.block(6, 6, nrOfJoints, nrOfJoints));
+
+    Eigen::VectorXd genBiasJointTorques(m_genForces.tail(nrOfJoints));
+
+    Eigen::MatrixXd pseudoInverseH = H.completeOrthogonalDecomposition().pseudoInverse();
+
+    Eigen::VectorXd FTBaseAcc = FTranspose * baseAcceleration;
+    Eigen::VectorXd totalTorques = motorTorqueAfterGearbox - frictionTorques - genBiasJointTorques + tauExt
+            + FTBaseAcc;
```

Comment:
Note that here the method is still allocating memory. ``FTranspose``, ``H``, ``genBiasJointTorques``, ``pseudoInverseH``, ``FTBaseAcc`` and ``totalTorques`` would need to be class members. In this way, their memory is preserved across different calls of the ``inverseDynamics`` method.

Reply:
I changed it :)